src/index.spec.js

```diff
@@ -1,27 +1,222 @@
+import mongodb from 'mongodb';
 import Mongo from './index';
 
+let db;
+const user = {
+  username: 'johndoe',
+  email: 'john@doe.com',
+  password: 'toto',
+  profile: {},
+};
+
+function createConnection(cb) {
+  const url = 'mongodb://localhost:27017/accounts-mongo-tests';
+  mongodb.MongoClient.connect(url, (err, dbArg) => {
+    db = dbArg;
+    cb(err);
+  });
+}
+
+function dropDatabase(cb) {
+  db.dropDatabase((err) => {
+    if (err) return cb(err);
+    return cb();
+  });
+}
+
+function closeConnection(cb) {
+  dropDatabase((err) => {
+    db.close();
+    if (err) return cb(err);
+    return cb();
+  });
+}
+
 describe('Mongo', () => {
+  beforeAll(createConnection);
+
   describe('#constructor', () => {
     it('should have default options', () => {
-      const mongo = new Mongo();
+      const mongo = new Mongo(db);
       expect(mongo.options).toBeTruthy();
     });
 
     it('should overwrite options', () => {
-      const mongo = new Mongo({
+      const mongo = new Mongo(db, {
         collectionName: 'users-test',
       });
       expect(mongo.options).toBeTruthy();
       expect(mongo.options.collectionName).toEqual('users-test');
     });
+
+    it('should throw with an invalid database connection object', () => {
+      try {
+        new Mongo(); // eslint-disable-line no-new
+        throw new Error();
+      } catch (err) {
+        expect(err.message).toBe('A valid database connection object is required');
+      }
+    });
+  });
+
+  describe('setupIndexes', () => {
+    it('should create indexes', async () => {
+      const mongo = new Mongo(db);
```

Comment:
All tests except of one uses this `mongo` instance. Meaning that you can save that instance on the file scope (as you did with `db` instance.